src/app/public/modules/dropdown/dropdown-menu.component.ts

```diff
@@ -284,4 +282,130 @@ export class SkyDropdownMenuComponent implements AfterContentInit, OnDestroy {
       return (i === index);
     });
   }
+
+  private sendMessage(type: SkyDropdownMessageType): void {
+    this.dropdownComponent.messageStream.next({ type });
+  }
+
+  private addEventListeners(): void {
+    const dropdownMenuElement = this.elementRef.nativeElement;
+
+    Observable
+      .fromEvent(dropdownMenuElement, 'click')
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((event: MouseEvent) => {
+        const selectedItem = this.menuItems.find((item: SkyDropdownItemComponent, i: number) => {
+          const found = (item.elementRef.nativeElement.contains(event.target));
+
+          if (found) {
+            this.menuIndex = i;
+            this.menuChanges.next({
+              activeIndex: this.menuIndex
+            });
+          }
+
+          return found;
+        });
+
+        /* istanbul ignore else */
+        if (selectedItem) {
+          this.menuChanges.next({
+            selectedItem
+          });
+        }
+      });
+
+    Observable
+      .fromEvent(dropdownMenuElement, 'keyup')
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((event: KeyboardEvent) => {
+        const key = event.key.toLowerCase();
+
+        /*tslint:disable-next-line:switch-default*/
+        switch (key) {
+          case 'escape':
```

Comment:
The ideal setup is `keyup` because it reflects a deliberate action from the user (i.e., `keydown` would fire if you pressed a key down but didn't remove your finger from the key). However, for some keys (like arrow keys and tab), JavaScript performs native functionality on the `keydown` action, which we need to short-circuit to perform our own actions. (These events should be copied from the original implementation, for the most part.)